### PR TITLE
JavaScript. Homework 6

### PR DESCRIPTION
**Pull request by VioletaScvortova**
Hello, Ivan! 

This homework was the hardest one for me. Unfortunately, to complete each task, I had to google and look for examples on the Internet. As you can see from the open tabs on the screen, I asked Google a lot of questions :)))

With grief in half, I completed 5 tasks out of 7. 

Tasks 3 and 6: I think I understood the logic of the example you gave us at the lecture with 'do...while' loop, but I didn't manage to apply it to the homework

JS_ Homework 6

### CODE REVIEW

**Comment by IvanCeban** [comment]
This is totally fine to use Google :)
I have tons of tabs opened every day for each task. So keep calm, you're on a correct track

**Comment by VioletaScvortova** [comment]
@IvanCeban hello! I`ve added some updates to tasks 3, 5, 6. 
image

File: JavaScript homeworks/homework-6.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
+//Loops task 1//
+let PersonDescription = ['name: John', 'age: 30', 'city: New-York'];
```
Comment (IvanCeban) [inline JavaScript homeworks/homework-6.js:2]: this was supposed to be an object, not array

File: JavaScript homeworks/homework-6.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+name: John
+age: 30
+city: New-York 
+*/
+
+
+//Loops task 2//
+function keyInObject(obj, keyToFind) {
+    for (let key in obj) {
+        if (key == keyToFind) {
```
Comment (IvanCeban) [inline JavaScript homeworks/homework-6.js:23]: better use strict comparison here

File: JavaScript homeworks/homework-6.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    prop2: 'World',
+    prop3: '!'
+}
+
+console.log(combineTheWords(obj));
+
+
+//Loops task 4//
+function countNumberOfVowels(string) {
+    let vowels = ['a', 'A', 'e', 'E', 'i', 'I', 'o', 'O', 'u', 'U'];
```
Comment (IvanCeban) [inline JavaScript homeworks/homework-6.js:74]: no need to check the uppercase vowels. Better convert the incoming string to lowercase and compare it with lowercase vowels only

File: JavaScript homeworks/homework-6.js [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+console.log(result); // 8
+
+
+//Loops task 5//
+//v1
+function calculateAverageScore(students) {
+    let average = {};
+
+    students.forEach(s => {
+        let averageScore = s.marks.reduce((a, i) => a + i) / s.marks.length;
```
Comment (IvanCeban) [inline JavaScript homeworks/homework-6.js:99]: I highly encourage you to use only the tools that you understand. 
If you understand how you got the correct result with reduce, good for you and keep it as it is. 
Bu if it is not clear for you, better use the loops that we already learned

**Review by IvanCeban — CHANGES_REQUESTED** [review]

File: JavaScript homeworks/homework-6.js (hunk 1 above)
Comment (VioletaScvortova) [reply JavaScript homeworks/homework-6.js]: Added changes, thank you!

File: JavaScript homeworks/homework-6.js (hunk 2 above)
Comment (VioletaScvortova) [reply JavaScript homeworks/homework-6.js]: Done!

File: JavaScript homeworks/homework-6.js (hunk 4 above)
Comment (VioletaScvortova) [reply JavaScript homeworks/homework-6.js]: +

File: JavaScript homeworks/homework-6.js (hunk 3 above)
Comment (VioletaScvortova) [reply JavaScript homeworks/homework-6.js]: added "string = string.toLowerCase();"

**Review by IvanCeban — APPROVED** [review]

**Pull request merged**